contracts/test/StakeTable.t.sol

```diff
@@ -0,0 +1,272 @@
+
+// Token contract
+import { ExampleToken } from "../src/ExampleToken.sol";
+
+// Target contract
+import { StakeTable as S } from "../src/StakeTable.sol";
+
+contract StakeTable_register_Test is Test {
+    event Registered(bytes32, uint64, AbstractStakeTable.StakeType, uint256);
+
+    S public stakeTable;
+    ExampleToken public token;
+    LightClient public lightClientContract;
+    uint256 constant INITIAL_BALANCE = 1_000;
+    address exampleTokenCreator;
+
+    function genClientWallet(address sender)
+        private
+        returns (BN254.G2Point memory, EdOnBN254.EdOnBN254Point memory, BN254.G1Point memory)
+    {
+        // Generate a BLS signature and other values using rust code
+        string[] memory cmds = new string[](3);
+        cmds[0] = "diff-test";
+        cmds[1] = "gen-client-wallet";
+        cmds[2] = vm.toString(sender);
+
+        bytes memory result = vm.ffi(cmds);
+        (
+            uint256 blsSigX,
+            uint256 blsSigY,
+            uint256 blsVKx0,
+            uint256 blsVKx1,
+            uint256 blsVKy0,
+            uint256 blsVKy1,
+            uint256 schnorrVKx,
+            uint256 schnorrVKy
+        ) = abi.decode(
+            result, (uint256, uint256, uint256, uint256, uint256, uint256, uint256, uint256)
+        );
+
+        return (
+            BN254.G2Point(blsVKx1, blsVKx0, blsVKy1, blsVKy0), // blsVK
+            EdOnBN254.EdOnBN254Point(schnorrVKx, schnorrVKy), // schnorrVK
+            BN254.G1Point(blsSigX, blsSigY) // sig
+        );
+    }
+
+    function setUp() public {
+        exampleTokenCreator = makeAddr("tokenCreator");
+        vm.prank(exampleTokenCreator);
+        token = new ExampleToken(INITIAL_BALANCE);
+
+        LightClient.LightClientState memory genesis = LightClient.LightClientState({
+            viewNum: 0,
+            blockHeight: 0,
+            blockCommRoot: 0,
+            feeLedgerComm: 0,
+            stakeTableBlsKeyComm: 0,
+            stakeTableSchnorrKeyComm: 0,
+            stakeTableAmountComm: 0,
+            threshold: 0
+        });
+        lightClientContract = new LightClient(genesis,10);
+        address lightClientAddress = address(lightClientContract);
+        stakeTable = new S(address(token),lightClientAddress);
+    }
+
+    function test_RevertWhen_UsingRestakeToken() external {
+        uint64 depositAmount = 10;
+        uint64 validUntilEpoch = 5;
+
+        (
+            BN254.G2Point memory blsVK,
+            EdOnBN254.EdOnBN254Point memory schnorrVK,
+            BN254.G1Point memory sig
+        ) = genClientWallet(exampleTokenCreator);
+
+        // Throw "Restaking not implemented" error
+        vm.expectRevert(S.RestakingNotImplemented.selector);
+        stakeTable.register(
+            blsVK,
+            schnorrVK,
+            depositAmount,
+            AbstractStakeTable.StakeType.Restake,
+            sig,
+            validUntilEpoch
+        );
```

Comment:
nit: this is a take-or-leave suggestion
instead of hardcoding these two variables, I would fuzz them: 
```solidity
   function testFuzz_RevertWhen_UsingRestakeToken(uint64 depositAmount, uint64 validUntilEpoch) external {
       uint64 curEpoch = stakeTable.currentEpoch();
       depositAmount = bound(depositAmount, 1, INITIAL_BALANCE);
       validUntilEpoch = bound(validUntilEpoch, curEpoch, curEpoch + 10);
     }     
```

personally i find this test much more convincing if I'm a security auditor.

Reply:
Done in 3a3ec04.